client/app/bundles/course/courses/components/forms/CourseInvitationCodeForm.tsx

```diff
@@ -0,0 +1,83 @@
+import { defineMessages, injectIntl, WrappedComponentProps } from 'react-intl';
+import { FC, useState } from 'react';
+import { Button, TextField } from '@mui/material';
+import { toast } from 'react-toastify';
+import { useDispatch } from 'react-redux';
+import { AppDispatch } from 'types/store';
+import { getCourseId } from 'lib/helpers/url-helpers';
+import { getRegistrationURL } from 'lib/helpers/url-builders';
+import { sendNewRegistrationCode } from '../../operations';
+
+interface Props extends WrappedComponentProps {}
+
+const translations = defineMessages({
+  placeholder: {
+    id: 'course.courses.show.invitation.placeholder',
+    defaultMessage: 'Invitation code',
+  },
+  registerButton: {
+    id: 'course.courses.show.invitation.registerButton',
+    defaultMessage: 'Register',
+  },
+  codeSubmitFailure: {
+    id: 'course.courses.show.invitation.codeSubmitFailure',
+    defaultMessage: 'Your code is incorrect',
+  },
+  emptyCodeFailure: {
+    id: 'course.courses.show.invitation.emptyCodeFailure',
+    defaultMessage: 'Please enter an invitation code',
+  },
+});
+
+const CourseInvitationCodeForm: FC<Props> = (props) => {
+  const { intl } = props;
+
+  const [code, setCode] = useState('');
+
+  const dispatch = useDispatch<AppDispatch>();
+
+  const handleSubmit = (): Promise<void> => {
+    if (code.trim() === '') {
+      toast.error(intl.formatMessage(translations.emptyCodeFailure));
+      return new Promise<void>(() => {});
+    }
+    const registrationLink = getRegistrationURL(getCourseId());
+    const data = new FormData();
+    data.append('registration[code]', code);
+    return dispatch(sendNewRegistrationCode(registrationLink, data))
+      .then(() => {
+        window.location.reload();
```

Comment:
Can i check why do we reload here? I think we can just make a json api call to show index instead and replace the data in the store?

Reply:
Need to re-render the entire page for the rails stuff to render (breadcrum, sidebar)

Reply:
got it thanks!